### PR TITLE
Homepage

### PR DESCRIPTION
**Pull request by Tanay-J**
PR features
- General setup (folder structure, routes etc-) for react app using mockBee backend
- Homepage & Navigation components for the bookstore

Most of the code has already been reviewed in previous e-commerce project.

Files to be reviewed (11)
- App.js
- Banner.jsx
- Featured.jsx
- GenreCategories.jsx
- Homepage.module.css
- SpecialCategories.jsx
- Footer.jsx
- Navbar.jsx
- Topbar.jsx
- useCategories.jsx
- useProducts.jsx

image

### CODE REVIEW

File: src/components/homepage/Homepage.module.css [hunk 1]
```diff
@@ -0,0 +1,75 @@
+@import url(/src/styles/utilities.css);
+
+.main_img_container {
+  position: relative;
+  overflow: hidden;
+}
+
+.main_img {
+  width: 100%;
+  max-height: 450px;
```
Comment (theprogrammerinyou) [inline src/components/homepage/Homepage.module.css:10]: Better to use rem for larger px values.

File: src/components/homepage/SpecialCategories.jsx [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+import { useCategories } from "../../hooks";
+
+const SpecialCategories = () => {
+  const { categories } = useCategories();
+  return (
+    <section className="wrapper mx-auto my-xl">
+      <h6 className="h6 text-dark text-center">Special Categories</h6>
+      <div className={`${styles.categories} my-m`}>
+        {categories.map((category) => {
+          return (
```
Comment (theprogrammerinyou) [inline src/components/homepage/SpecialCategories.jsx:11]: add key prop whenever using map in jsx.

**Review by theprogrammerinyou — COMMENTED** [review]
Well written code, but please avoid putting too many commits into one PR. You can also keep routes in different folder, just a suggestion not necessary, upto you. Thank you. :)

File: src/App.js [hunk 3]
```diff
@@ -1,43 +1,24 @@
...
+        <Route path="/login" element={<Login />}></Route>
+        <Route path="/signup" element={<Signup />}></Route>
+        <Route path="/products" element={<Products/>}></Route>
+        <Route path="/products/fiction" element={<Products category='Fiction'/>}></Route>
+        <Route path="/products/nonfiction" element={<Products category='Non-Fiction'/>}></Route>
+        <Route path="/products/selfhelp" element={<Products category='Self-Help'/>}></Route>
+        <Route path="/products/boxset" element={<Products category='Box-set'/>}></Route>
+        <Route path="/products/justin" element={<Products category='Just In'/>}></Route>
+        <Route path="/products/popular" element={<Products category='Popular'/>}></Route>
+      </Routes>
```
Comment (dikshadhote) [inline src/App.js]: Use separate Router file and then import in App

**Review by dikshadhote — APPROVED** [review]
For all folder structure set up, you can directly do that in dev. Please raise PR for specific component changes.
Also include deployment preview link.so, that reviewer can see the UI and inspect it.
Homepage only needs categories data.so, you could have used separate PR for products.
Overall the code is neat and clean but work on above things.

File: src/App.js (hunk 3 above)
Comment (Tanay-J) [reply src/App.js]: Thank you for the suggestions, Diksha. Will keep these in mind while raising PRs in future.

File: src/components/homepage/SpecialCategories.jsx (hunk 2 above)
Comment (Tanay-J) [reply src/components/homepage/SpecialCategories.jsx:11]: Done.

**Pull request merged**